python: remove unneeded name attributes, remove rec, make afew an application

flokli opened the pull request:
###### Motivation for this change
As I found out, it's not necessary to specify the `name` for python packages if we set `pname` and `version`. Additionally, get rid of the `rec`, and change the `afew` package to be a python application (what it is, it's not a library).

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

flokli committed:
bpython: don't set name attribute anymore

flokli committed:
curtsies: don't set name attribute anymore

flokli committed:
guzzle_sphinx_theme: don't set name attribute anymore

flokli committed:
pyte: don't set name attribute anymore

flokli committed:
twilio: don't set name attribute anymore

flokli committed:
afew: it's an application, not a python module

Also, remove name attribute.
Build with python 3 by default.

dezgeg commented on pkgs/development/python-modules/curtsies/default.nix:
```diff
@@ -1,10 +1,12 @@
 { stdenv, buildPythonPackage, fetchPypi, blessings, mock, nose, pyte, pytest, wcwidth }:
 
-buildPythonPackage rec {
+let
```
Why the separate let block?

flokli commented:
to avoid having to use rec in the set below

FRidh commented on pkgs/applications/networking/mailreaders/afew/default.nix:
```diff
@@ -0,0 +1,32 @@
+    sha256 = "121w7bd53xyibllxxbfykjj76n81kn1vgjqd22izyh67y8qyyk5r";
+  };
+
+  propagatedBuildInputs = with pythonPackages; [
```
Setuptools_scm is a build time input

FRidh commented on pkgs/applications/networking/mailreaders/afew/default.nix:
```diff
@@ -0,0 +1,32 @@
+    setuptools_scm pythonPackages.notmuch chardet
+  ] ++ stdenv.lib.optional (!pythonPackages.isPy3k) subprocess32;
+
+  SETUPTOOLS_SCM_PRETEND_VERSION = "${version}";
```
preBuild is this likely a better place for this one

flokli replied:
seems like this isn't required anymore at all, as we're fetching from pypi - and setuptools_scm is able to figure out the version on its own

FRidh commented on pkgs/applications/networking/mailreaders/afew/default.nix:
```diff
@@ -0,0 +1,32 @@
+
+  postInstall = ''
+    wrapProgram $out/bin/afew \
+      --prefix LD_LIBRARY_PATH : ${notmuch}/lib
```
Can this be fixed by hardcoring the path to the lib. Using LD_LIBRARY_PATH should be prevented

flokli replied:
It's already hardcoded into `pythonPackages.notmuch`. Will strip this.

FRidh commented on pkgs/applications/networking/mailreaders/afew/default.nix:
```diff
@@ -0,0 +1,32 @@
+      --prefix LD_LIBRARY_PATH : ${notmuch}/lib
+  '';
+
+  meta = with stdenv.lib; {
```
License?

FRidh commented:
Why do you want to prevent the recursive set?

flokli commented:
@FRidh I was under the impression `rec` is costly and should be avoided if not really needed. Seems I'm mistaken?

Thanks for reviewing the other things, seems like this slipped through in previous PRs. WIll update soonish.

flokli committed:
afew: make setuptools_scm a buildInput

flokli committed:
afew: don't set SETUPTOOLS_SCM_PRETEND_VERSION anymore

we fetch from pypi, with version info properly set, so this shouldn't be
an issue.

flokli committed:
afew: set license

flokli committed:
afew: add notmuch binary to PATH

it seems afew calls "notmuch new" in MailMover.py

flokli committed:
afew: don't set LD_LIBRARY_PATH

the library is already hardcoded pythonPackages.notmuch's globals.py

FRidh commented:
There is no need to avoid `rec` here.

flokli commented:
@FRidh Updated.

FRidh commented:
@flokli can you squash the `afew` changes.

FRidh commented:
Included changes in https://github.com/NixOS/nixpkgs/pull/34077.
